ci/Jenkinsfile-databricks

```diff
@@ -0,0 +1,246 @@
+#!/usr/bin/groovy
+@Library('test-shared-library') _
+import groovy.json.JsonSlurperClassic
+import java.nio.file.Files
+import java.nio.file.Paths
+
+properties(
+        [
+                pipelineTriggers([cron('H 16 * * *')]),
+                buildDiscarder(logRotator(numToKeepStr: '10'))
+        ]
+)
+
+def readPropertiesFile(file) {
+    def properties = [:]
+    readFile(file).split("\n").each { line ->
+        if (!line.startsWith("#")) {
+            def splits = line.split("=")
+            if (splits.length == 1) {
+                properties[splits[0]] = ""
+            } else {
+                properties[splits[0]] = splits[1]
+            }
+        }
+    }
+    return properties
+}
+
+static String getSparklingVersion(props) {
+    return "${props['version'].replace("-SNAPSHOT", "")}"
+}
+
+def startCluster(dbcVersion, sparkMajorVersion) {
+    def jsonOutput = sh(script: """
+    databricks clusters create --json '{
+        "num_workers": 3,
+        "cluster_name": "SparklingWaterTest-${sparkMajorVersion}",
+        "spark_version": "${dbcVersion}",
+        "spark_conf": {},
+        "aws_attributes": {
+            "first_on_demand": 4,
+            "availability": "SPOT_WITH_FALLBACK",
+            "zone_id": "us-east-1d",
+            "instance_profile_arn": null,
+            "spot_bid_price_percent": 100,
+            "ebs_volume_type": "GENERAL_PURPOSE_SSD",
+            "ebs_volume_count": 3,
+            "ebs_volume_size": 100
+        },
+        "node_type_id": "m4.large",
+        "ssh_public_keys": [],
+        "custom_tags": {},
+        "spark_env_vars": {},
+        "autotermination_minutes": 120,
+        "init_scripts": []
+    }'
+""", returnStdout: true).trim()
+    def jsonSlurper = new JsonSlurperClassic()
+    def cfg = jsonSlurper.parseText(jsonOutput)
+    return cfg["cluster_id"]
+}
+
+def waitForCluster(clusterId) {
+    def ready = false
+    while(!ready) {
+        def json = sh(script: "databricks clusters get --cluster-id ${clusterId}", returnStdout: true).trim()
```

Comment:
What happens if the creation of the cluster fails?

Reply:
Tests fail

Reply:
I have added more explicit handling in case something wrong happens during start of the cluster